### ISSUE TITLE
Block events with excessively large arguments

### ISSUE THREAD

**Issue by nickdaugherty** [Automattic/Cron-Control#159]
For stability, cron should not accept events with arguments of excessive size, because this causes problems with caching, db load, network transfer, all sorts of fun stuff.

Usually these types of cron events are storing full objects as arguments, instead of ids - like terms for example.

There should never be a situation where a cron event contains 400kb of data 😱 

Need to find the right cutoff, but 10k might be a good starting point?

**Comment by nickdaugherty** [comment]
Need to think about how to surface failures - can't silently fail.

**Comment by ethitter** [comment]
> Need to think about how to surface failures - can't silently fail.

Indeed. Not scheduling the event will essentially create a race condition as the code continually tries to do so, as we saw with the "skip events with no callbacks" feature.

To avoid that, it might make sense to schedule the event but drop the arguments, replacing them with some flag to delete the event without running. As long as the action, timestamp, and instance are properly set, Cron Control shouldn't keep rescheduling a bad event.

**Comment by nickdaugherty** [comment]
Good catch - we should definitely do that to prevent repeated re-scheduling.

**Comment by WPprodigy** [comment]
This should be a pretty non-issue when we can stop relying on the `cron` option completely. Once the new core filters are implemented, almost all use cases for `cron` are gone (except backwards compatibility w/ custom plugin code).

There are only two places WP core still uses `_get_cron_array()` w/o it being filterable beforehand, but these are rare and not super mission critical (site health page, media table reading a `upgrader_scheduled_cleanup` job).

tl;dr, we can "soon" make the `cron` option not auto-loaded. It can still return the combined bucketed data when it's called manually (and perhaps we throw deprecating notice/warning)

**Comment by WPprodigy** [comment]
W/ the upcoming changes, this should mostly be a non-issue. If custom code is still calling `get_option('cron')`, at most we cache in "buckets" of 100 events.

I'm thinking we can start w/ just throwing a PHP warning during event (re)scheduling if the args is rather significantly large, that way it'll be notable when debugging application performance concerns. But we won't run into any troubles by deviating from expected functionality.

**Comment by WPprodigy** [comment]
Started logging for this here: https://github.com/Automattic/Cron-Control/pull/237. Pretty gracious on the limit, decided on something that will enable to cache up to 500 events per cache key. Seems reasonable.
